src/dmd/typesem.d

```diff
@@ -3139,7 +3139,7 @@ private extern(C++) final class DotExpVisitor : Visitor
 
             e = new TupleExp(e.loc, e0, exps);
             Scope* sc2 = sc.push();
-            sc2.flags = sc.flags | SCOPE.noaccesscheck;
+            sc2.flags |= global.params.vsafe ? SCOPE.onlysafeaccess : SCOPE.noaccesscheck;
```

Comment:
I don't think this should be tied to DIP 1000 (`global.params.vsafe`). DIP 1000 is about `scope`. But this has nothing to do with `scope`.

Reply:
In other precendents breaking changes were introduced with `-dip1000` as it's effectively unused, e.g. https://github.com/dlang/dmd/pull/8036 (there were more in the last weeks, but I can't find them on the first try).
I guess eventually dip1000 might evolve in a sublanguage :/